### ISSUE TITLE
Pinpoint not sampling all request in spite of setting profiler.sampling.rate=1 in config

### ISSUE THREAD

**Issue by mohit24** [pinpoint-apm/pinpoint#7847]
**Pinpoint Version**: 2.2.2
**Java Version:** 1.8
**Spring Boot version:** 2.3.4.RELEASE

We are using Spring Webflux with netty server.

**Pinpoint config snippet:**
 
profiler.sampling.enable=true
profiler.sampling.rate=1

**Command to run spring boot application:** 
java -jar -Dlogs_location=/home/dev/logs -javaagent:pinpoint-bootstrap-2.2.2.jar -Dpinpoint.applicationName=Perf-Sample -Dpinpoint.agentId=app1-perf ct-web-0.0.1-SNAPSHOT.jar --spring.profiles.active=perf

Only 1 transaction (SN) is being sampled out of 22 transaction (UN). Attaching the screenshot for reference.
 
![Screenshot 2021-05-04 at 8 23 11 PM](https://user-images.githubusercontent.com/10355812/117025057-30886e80-ad18-11eb-926f-d32885642b4d.png)
![Screenshot 2021-05-04 at 8 23 36 PM](https://user-images.githubusercontent.com/10355812/117025070-33835f00-ad18-11eb-8a74-44a827231f05.png)

**Comment by emeroad** [comment]
Sampling is determined by the root node.
If the root node decides not to sample, the child node follow it.

**Comment by mohit24** [comment]
@emeroad I didn't get what you said. Can you please help me to understand? What else do I need to put in config to get all the request tracked?

**Comment by emeroad** [comment]
pinpoint uses head-based sampling.

Take a look at this article
https://lightstep.com/blog/the-truth-about-sampling-and-distributed-tracing/

**Comment by mohit24** [comment]
@emeroad Thanks for the article. I got the idea about head-based sampling. But my problem is not related to this I guess. In my case, there is no root node and child node. In my scenario, there is only one service and I want to track the request/response of only this service. If I have set the profiler.sampling.enable=true, profiler.sampling.rate=1 in config, I should be able to see all the requests landing on my service which is not happening.
